src/BlockTypesController.php

```diff
@@ -152,8 +153,12 @@ public function block_hydration_wrapper( $block_content, $block, $instance ) {
 			return $block_content;
 		}
 
-		$block_type       = $instance->block_type;
-		$attr_definitions = isset( $block_type ) ? $block_type->attributes : [];
+		$block_type = $instance->block_type;
+		if ( ! $block_type ) {
+			// We might use a better way to flag it here.
+			wc_add_notice( 'You need to make a server registration of ' . $block['blockName'], 'error' );
```

Comment:
Maybe we can use [`_doing_it_wrong()`](https://developer.wordpress.org/reference/functions/_doing_it_wrong/) here? Although that's supposed to flag a function that's invoked in the wrong way; not totally sure that applies here 🤔 

Although maybe it's arguable that the `render_block` hook should be called with 3 args? 🤔

Reply:
As @tarhi-saad just told me, they server-registered all cart inner blocks in [this PR](https://github.com/woocommerce/woocommerce-blocks/pull/6737) in order to fix a translation issue.

I will try to update the repo. So that way it would be fixed. Then we can discuss about adding a flag or not 😆

Reply:
Ah! Of course, he told me that too 😆 I'd just forgotten about it 😅